java-checks-testkit/src/main/java/org/sonar/java/checks/verifier/CheckVerifier.java

```diff
@@ -114,13 +114,28 @@ static CheckVerifier newVerifier() {
   /**
    * Defines the java version syntax to be used for the verification. Usually used when the code of the
    * test files target explicitly a given version (eg. java 7), where a particular syntax/API has been introduced.
+   * Preview features for the specified java version will be disabled by default, use {@link CheckVerifier#withJavaVersion(int, boolean)} 
+   * to enable or disable preview features associated with the specified java version.
    *
    * @param javaVersionAsInt defines the java language syntax version to be considered during verification, provided as an integer.
    * For instance, for Java 1.7, use '7'. For Java 12, simply '12'.
    *
    * @return the verifier configured to consider the provided test file(s) as following the syntax of the given java version
    */
   CheckVerifier withJavaVersion(int javaVersionAsInt);
+  
+  /**
+   * Defines the java version syntax to be used for the verification. Usually used when the code of the
+   * test files target explicitly a given version (eg. java 7), where a particular syntax/API has been introduced.
+   *
+   * @param javaVersionAsInt defines the java language syntax version to be considered during verification, provided as an integer.
+   * For instance, for Java 1.7, use '7'. For Java 12, simply '12'.
+   * 
+   * @param enablePreviewFeatures defines if preview features from the specified java version should be enabled or not
+   *
+   * @return the verifier configured to consider the provided test file(s) as following the syntax of the given java version
+   */
+  CheckVerifier withJavaVersion(int javaVersionAsInt, boolean enablePreviewFeatures);
```

Comment:
I'm not sure this is a good approach.
1. ECJ supports preview features only with its latest version of the language. So for us, we have the same behavior: the preview feature is supported when version == latest java version supported. In our case, 19. So allowing people to enable it for other than the latest is useless. It should not work. Did you test this scenario?
2. I believe it's not great to have a "flag" to enable the preview features. I would prefer two different methods. One without (implicitly), and one with the preview feature enabled.
3. Since only the latest version supports preview features, what about a new method `withPreviewFeatureEnabled()` without parameters? This would either force the latest version or check that the latest has been configured and crash if not, and eventually enable preview features.